src/custom-lists/background/storage.ts

```diff
@@ -0,0 +1,191 @@
+import { FeatureStorage } from '../../search/search-index-new/storage'
+import { ListObject, PageObject } from './types'
+
+const COLLECTION_NAME = 'customLists'
+const PAGE_LIST_ENTRY = 'pageListEntries'
+
+// TODO: Add typings for the class
+export default class CustomListStorage extends FeatureStorage {
+    constructor({ storageManager }) {
+        super(storageManager)
+
+        this.storageManager.registerCollection(COLLECTION_NAME, {
+            // different version for adding a new table.
+            version: new Date(2018, 6, 12),
+            fields: {
+                id: { type: 'string', pk: true },
+                name: { type: 'text' },
+                isDeletable: { type: 'binary' },
+                isNestable: { type: 'binary' },
+                createdAt: { type: 'datetime' },
+            },
+            indices: [
+                { field: 'id', pk: true, autoInc: true },
+                { field: 'name', unique: true },
+                { field: 'isDeletable' },
+                { field: 'isNestable' },
+                { field: 'createdAt' },
+            ],
+        })
+
+        this.storageManager.registerCollection(PAGE_LIST_ENTRY, {
+            // different version for adding a new table.
+            version: new Date(2018, 6, 12),
+            fields: {
+                listId: { type: 'string' },
+                pageUrl: { type: 'string' },
+                fullUrl: { type: 'string' },
+                createdAt: { type: 'datetime' },
+            },
+            indices: [
+                { field: ['listId', 'pageUrl'], pk: true },
+                { field: 'listId' },
+                { field: 'pageUrl' },
+            ],
+        })
+    }
+
+    // Return all the list in the DB
+    //  TODO: Use pagination if required and decide correct types
+    async fetchAllList({ query = {}, opts = {} }: { query?: any; opts?: any }) {
+        const x = await this.storageManager.findAll(COLLECTION_NAME, query, opts)
+        return this.changeListsbeforeSending(x, [])
+    }
+
+    // Takes the list as they come from Db and does some pre-processing before sending.
+    async changeListsbeforeSending(lists: object[], pageEnteries: object[]) {
+        const mappedLists = lists.map((list: ListObject) => {
+            const page = pageEnteries.find(({ listId }: PageObject) => listId === list.id)
+            delete list["_name_terms"]
+            return {
+                ...list,
+                // Set an empty pages array for manupulating in redux store.
+                pages: [],
+                active: Boolean(page),
+            }
+        })
+        return mappedLists
+    }
+
+    async fetchListById(id: Number) {
```

Comment:
Be careful with TS primitive types. `Number` is different to `number` (correct usage here). More info here:
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#general-types

I also don't think this method takes into consideration the case when the first query (looking up the list for the given `id`) doesn't match any lists. I think it should return `undefined` or some clear behaviour if there is no matching list (`insertPageToList` method depends on this behaviour)